code/game/machinery/computer/atmos_controllers.dm

```diff
@@ -506,10 +506,15 @@ GLOBAL_LIST_EMPTY(gas_sensors)
 	circuit = /obj/item/circuitboard/atmoscontrol
 	req_access = list(ACCESS_ATMOSPHERICS)
 	var/datum/ui_module/atmos_control/atmos_control
+	var/parent_area_type
 
 /obj/machinery/computer/atmoscontrol/Initialize(mapload)
 	. = ..()
 	atmos_control = new(src)
+	var/area/machine_area = get_area(src)
+	parent_area_type = machine_area.get_top_parent_type()
+	atmos_control.parent_area_type = parent_area_type
+	atmos_control.z_level = z
```

Comment:
Is it possible that this machine could be moved, causing this to become invalid?

Reply:
You need to disassemble a console to move it, so it would reinitialize at the new location. I tested to make sure.